processing/src/main/java/io/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -179,7 +179,7 @@ private void waitForFutureCompletion(
     try {
       queryWatcher.registerQuery(query, future);
       final long timeout = QueryContexts.getTimeout(query);
-      if (timeout == 0) {
+      if (QueryContexts.isNoTimeout(timeout)) {
```

Comment:
Ah, sorry I missed this comment. Yeah, it looks good. I changed including ```ChainedExecutionQueryRunner```. It also looks simple and good.